packages/block-library/src/query/edit/inspector-controls/index.js

```diff
@@ -123,21 +123,18 @@ export default function QueryInspectorControls( props ) {
 	const showInheritControl =
 		isTemplate && isControlAllowed( allowedControls, 'inherit' );
 	const showPostTypeControl =
-		( ! inherit && isControlAllowed( allowedControls, 'postType' ) ) ||
-		! isTemplate;
+		! inherit && isControlAllowed( allowedControls, 'postType' );
```

Comment:
`inherit` is already being set to false if on singular content in [query-conent.js](https://github.com/WordPress/gutenberg/blob/trunk/packages/block-library/src/query/edit/query-content.js#L111), so we can remove the `isTemplate` check from these lines.

The only `isTemplate` check we need here is the one on line 124, as this decides whether to show the inherit control or not, and the intended behaviour is for this control to only show when on non-singular content (and if the control is allowed).